examples/add_points_with_properties.py

```diff
@@ -45,11 +45,8 @@
     def toggle_point_annotation(viewer):
         selected_points = list(points_layer.selected_data)
         if len(selected_points) > 0:
-            good_point = points_layer.properties['good_point']
+            props = points_layer.properties
+            good_point = props['good_point']
             good_point[list(selected_points)] = ~good_point[list(selected_points)]
-            points_layer.properties['good_point'] = good_point
-
-            # we need to manually refresh since we did not use the Points.properties setter
-            # to avoid changing the color map if all points get toggled to the same class,
-            # we set update_colors=False (only re-colors the point using the previously-determined color mapping).
-            points_layer.refresh_colors(update_color_mapping=False)
+            props['good_point'] = good_point
+            points_layer.properties = props
```

Comment:
This is an example of what I am talking about the `Points.refresh_colors()`. See how in the original example, we set the new property values with  points_layer.properties['good_point'] = good_point` and then have to manually refresh the colors because we didn't use the `Points.properties` setter.

Reply:
I added the `refresh_colors` method back in, so I think there are no breaking changes in this PR.

Reply:
currently going through this PR.  but wanted to note: if I select a point (or multiple), and change something like edge/face color, it immediately changes for the _first_ time I change it... but then if I change it again, it doesn't update the color. (however, if I then drag the point(s) (or even any other point) it updates the color).  (master updates even on later changes).  Must be related to `refresh_colors` but I'm still catching up

Reply:
https://user-images.githubusercontent.com/1609449/107149023-65a54080-6924-11eb-83d8-37ea2b3802b8.mov

Reply:
Thanks for the example, @tlambert03. I think this is related to the issue I am having with the events (see [this comment](https://github.com/napari/napari/pull/2204#discussion_r571615701)). I will see if I can fix the event emission issue first and then see if that fixes this issue.